File: Nivedita/Array/Move all negative elements to end .cpp
class Solution{
    public:
    void segregateElements(int arr[],int n)
    {
     vector<int>p,q;
     for(int i=0;i<n;i++)
     {
         if(arr[i]>0)
         {
             p.push_back(arr[i]);
         }
         else
         {
             q.push_back(arr[i]);
         }
     }
         for(int i=0;i<p.size();i++)
         {
             arr[i]=p[i];
         }
         for(int i=p.size();i<p.size()+q.size();i++)
         {
             arr[i]=q[i-p.size()];
         }
     }// Your code goes here
};

File: Saurabh Kr. Vigyananand/Array GFG/Move all negative elements to end.py
#User function Template for python3

class Solution:
    def segregateElements(self, arr, n):
        plst=[]
        nlst=[]
        for i in arr:
            if(i>=0):
                plst.append(i)
            else:
                nlst.append(i)
        j=0
        for i in range(len(plst)):
            arr[i]=plst[i]
            j+=1
        for i in range(len(nlst)):
            arr[j]=nlst[i]
            j+=1
            
        # Your code goes here

#{ 
#  Driver Code Starts
#Initial Template for Python 3


def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        ob=Solution()
        ob.segregateElements(a, n)
        print(*a)

        T -= 1


if __name__ == "__main__":
    main()





    
# } Driver Code Ends
